ledger/src/shred.rs

```diff
@@ -467,7 +473,25 @@ impl Shred {
         self.common_header.signature
     }
 
-    pub fn seed(&self) -> [u8; 32] {
+    pub fn seed(
+        &self,
+        leader_schedule_cache: &Arc<LeaderScheduleCache>,
+        bank_forks: &Arc<RwLock<BankForks>>,
+    ) -> [u8; 32] {
+        let root_bank = bank_forks.read().unwrap().root_bank();
```

Comment:
can this function take the `root_bank` as the argument instead of `bank_forks` so that we are not locking `bank_forks` for every single shred?

Reply:
changed